designer/server/src/main/scala/pl/touk/nussknacker/ui/component/ComponentGroupsPreparer.scala

```diff
@@ -1,70 +1,67 @@
 package pl.touk.nussknacker.ui.component
 
 import pl.touk.nussknacker.engine.api.component.ComponentGroupName
-import pl.touk.nussknacker.engine.api.process.ProcessingType
 import pl.touk.nussknacker.engine.definition.component._
 import pl.touk.nussknacker.engine.definition.component.defaultconfig.DefaultsComponentGroupName
 import pl.touk.nussknacker.engine.definition.model.ModelDefinition
 import pl.touk.nussknacker.engine.util.Implicits.RichTupleList
 import pl.touk.nussknacker.restmodel.definition._
-import pl.touk.nussknacker.ui.process.ProcessCategoryService
-import pl.touk.nussknacker.ui.security.api.LoggedUser
 
 import scala.collection.immutable.ListMap
 
-class ComponentGroupsPreparer(componentsGroupMapping: Map[ComponentGroupName, Option[ComponentGroupName]]) {
+object ComponentGroupsPreparer {
 
   def prepareComponentGroups(
-      user: LoggedUser,
       definitions: ModelDefinition[ComponentStaticDefinition],
-      processCategoryService: ProcessCategoryService,
-      processingType: ProcessingType
-  ): List[ComponentGroup] = {
+  ): List[UIComponentGroup] = {
     ComponentNodeTemplatePreparer
-      .componentNodeTemplatesWithGroupNames(user, definitions, processCategoryService, processingType)
+      .componentNodeTemplatesWithGroupNames(definitions)
+      .map(templateWithGroups =>
+        (templateWithGroups.originalGroupName, templateWithGroups.mappedGroupName) -> templateWithGroups.nodeTemplate
+      )
       .toGroupedMap
       .toList
-      .map { case (originalGroupName, nodeTemplates) =>
-        val nonHiddenMappedGroupName = componentsGroupMapping.getOrElse(originalGroupName, Some(originalGroupName))
-        (originalGroupName, nonHiddenMappedGroupName, nodeTemplates)
-      }
-      .flatMap { case (originalGroupName, nonHiddenMappedGroupName, nodeTemplates) =>
-        nonHiddenMappedGroupName.map { mappedGroupName =>
-          (originalGroupName, mappedGroupName, nodeTemplates)
-        }
-      }
+      // We sort based on the original group name. It might be tricky when someone instead of using component group
```

Comment:
I thought about that, but I came up with the conclusion, that it could be a few options to handle this:
- ordering in the configuration
- some hardcoded rules like these mentioned in the comment
So I decided to not do any more work around that now, and just to expose the logic that was already implemented. When we decide, which would be the final, clean approach, we change the domain model.